File: Java/src/main/java/com/gildedrose/items/Legendary.java
package com.gildedrose.items;

import com.gildedrose.Item;
import com.gildedrose.item_helpers.ItemHandler;
import com.gildedrose.item_helpers.ItemType;

public class Legendary implements ItemType {

    private final ItemHandler item;

    public Legendary(Item item) {
        this.item = new ItemHandler(item);
    }

    @Override
    public void updateQuality() {
        item.decrementSellInDate();
        item.setLegendaryQuality();
    }

}

File: csharp/Legendary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace csharp
{
    internal class Legendary : Item
    {
        public Legendary(string name, int sellIn, int quality) : base(name, sellIn, quality)
        {
            TypeOfItem = ItemType.Legendary;
        }

        protected override void QualityModifier()
        {
            //legendary items doesnt change!
            this.SellIn++;
        }
    }
}
